framework/python/src/common/session.py

```diff
@@ -100,6 +100,13 @@ def __init__(self, root_dir):
     self._load_config()
     self._load_profiles()
 
+    # Network information
+    self._ipv4_subnet = None

```

Comment:
Can we take networking info and put it into another core class? I don't think the session is the right place for it

Reply:
Since session is passed around to multiple services, I feel this might actually be the best place for all configurations to load into.  We also load the session first so making this the location that initiates all configurations first and then those services utilize those configurations seems the most efficient way to handle these things.

Reply:
Ok, to reduce the size of the session, could we not have a separate class like NetworkConfig or NetworkSession that is created and stored in the session?

Reply:
I think we could do something like that but make this a separate PR to work on the session alone since this is already a pretty large PR.